import "google/protobuf/descriptor.proto";

message ConfigFileOptions {
	optional string config_file_name = 1;
	optional string source_file_name = 2;
}

message ConfigMessageOptions {
	optional string template = 1; // {xx}-{yy}
}

message ConfigFieldOptions {
	optional string default = 1; // default value for encoded message string
	optional bool       map = 2;
	
	// for conf file
	optional string    note = 20;
	optional bool commented = 21;
	
	// for cmdline
	optional string       cmd = 40;
	optional string short_cmd = 41;
}

extend google.protobuf.FieldOptions {
    optional ConfigFieldOptions config_options = 60050;
}

extend google.protobuf.MessageOptions {
    optional ConfigMessageOptions config_message_options = 60050;
}

extend google.protobuf.FileOptions {
    optional ConfigFileOptions config_file_options = 60050;
}
